### ISSUE TITLE
V 0.5.7 Wishlist

### ISSUE THREAD

**Issue by BajczA475** [MAISRC/ggplotplus#13]
- [x] Easier way to add alt text.
- [x] Coaching to encourage doing so.
- [x] ~~If possible, some way to trigger auto-resizing as a function of the display window size.~~
- [x] ~~Institute a contrast check?~~
- [x] Experiment with whether focus mode could add transparency as an alternative mode.
- [x] See what it would take to create an icon array in ggplot2.
- [x] ~~See what it would take to create a value suppressing uncertainty palette.~~
- [x] ~~See what it would take to make a "fuzziness" visual channel.~~
- [ ] Consider a coaching message about error bars.
- [x] Change from discussing luminance and saturation to discussing lightness and chroma as the former are more the physical terms and the latter are more the perceptional ones.

**Comment by BajczA475** [comment]
So, it looks like it's easy to add alt text to ggplots via labs(alt = ...). This presumably means the alt text ends up in the final plot's labels. There is also a get_alt_text() function that can be used as a shortcut to check if a graph has alt text or not. A wrapper to labs(alt = ...) seems apt, as does a coaching message that checks for and reports on a missing alt text label.

**Comment by BajczA475** [comment]
To me, it looks like the waffle package ought to make an icon array: https://r-charts.com/part-whole/waffle-chart-ggplot2/

**Comment by BajczA475** [comment]
Here's an interesting post on VSUP with relevant code: http://lenkiefer.com/2018/05/02/what-s-up-vsup-that-s-what-s-up/. They're mapping transparency to uncertainty, but saturation might be cleaner and worth experimenting with.

**Comment by BajczA475** [comment]
Well, ChatGPT and its boundless optimism thinks a contrast-checking function would be possible, though it'd be quite hit or miss. https://chatgpt.com/share/6a8c983d-a7f4-83e9-a656-71709ab33c54. Something to maybe revisit in a later issue.

**Comment by BajczA475** [comment]
So, it terms of auto-resizing, this is indeed possible. For example, ChatGPT built this function:

.get_device_size = function(default = c(7.25, 5.95)) {

  dims = tryCatch(
    grDevices::dev.size("in"),
    error = function(e) { c(NA_real_, NA_real_)}
  )

  if(length(dims) != 2 || any(!is.finite(dims))) {
    dims = default
  }
  return(dims)
}

That would do something like that using dev.size(). It does work! But it acts based on the dimensions of the current graphics device at run time. For knitr docs and the Plots pane, that's sufficient. But for ggsave, it needs to wait until quite late in the dispatch to know what the final device dimensions will be, So, this would require a medium-complexity piping system to get intent up to that late a stage. Too much to bite off for this issue, but looks promising for future ones.

**Comment by BajczA475** [comment]
As for implementing a VSUP in R, I think that's doable...see this chat for some ideas. What it has already is pretty promising. https://chatgpt.com/share/6a8caa50-d5dc-83ea-bf9b-98326edd3941

**Comment by BajczA475** [comment]
After a lengthy debate on the merits of VSUPs, I'm not convinced that they are inherently the best solution to the problem. At extreme lightnesses, chroma disappears, so it would be impossible to discern uncertainty there if mapped to chroma. I also don't like the conflation of uncertainty with moderation--a highly uncertain value with an extreme point estimate is very different from a moderate value point in many contexts, and yet VSUPs conflate the two. Instead, I think recommending that transparency, size, or chroma (when not in conjunction with lightness) are good visual channels to use for conveying uncertainty because they convey vibes quickly, are intuitively linked with uncertainty, are already easy to effect, synergize with visual prominence, and don't distort or degrade the primary visual channels. They also do not require extra marks that might compound issues of overplotting, and they don't require readers to interpret more abstract channels like shape. So, no new features needed. This could be added as a bonus vignette to the guide, and there's some code to this effect in the Experimental.R file, but it's not a high-priority add.

**Comment by BajczA475** [comment]
I dunno that I like the idea of a fuzziness visual channel all that much. Its use case as a means of conveying uncertainty would negatively interact with its ability to show whatever primary channel was of interest, such as position or color.

**Comment by BajczA475** [comment]
I've substantially enhanced scale_focus_plus() to work with alpha, size, and linewidth as well, for when those might make more sense as focusing aesthetics. I probably need to dial in the defaults a bit more, but the guide, documentation, readme, tests, etc. are all updated accordingly.

**Comment by BajczA475** [comment]
I fine-tuned the default values just enough that I suspect this'll work ok. As a bonus, the scale_focus_plus calls stack, so focus via multiple simultaneous aesthetics is possible. This is maybe something the function should just accommodate.

**Comment by BajczA475** [comment]
I've generalized scale_focus_plus() to be able to take multiple compatible aes in the first argument.

**Comment by BajczA475** [comment]
It took a surprising amount of development but icon_array_plus() is now complete and will launch in version 0.5.7. I think that will conclude development for this version.
